plrust/src/allow_list.rs

```diff
@@ -212,6 +212,53 @@ pub enum Error {
     UnsupportedVersionReq(String),
 }
 
+// Tuple describes name of the dependency, version of the dependency, features of the dependecy that are enabled, whether default features enabled
+pub type AllowedDependencyTuple = (String, String, Vec<String>, bool);
+
+// Convert all the allowed dependencies into a vector of tuples
+pub fn get_allowed_dependencies_as_tuple_vector() -> Vec<AllowedDependencyTuple> {
+    let allowlist = load_allowlist().expect("Error loading dependency allow-list");
+    let mut tuples: Vec<AllowedDependencyTuple> = vec![];
+
+    for dependency in allowlist.values() {
+        let mut tuple: Vec<AllowedDependencyTuple> =
+            get_single_dependency_as_tuple_vector(dependency);
+        tuples.append(&mut tuple);
+    }
+    tuples
+}
+
+// Multiple version of a dependencies can be allowed. Create a tuple for each version of the dependency
+fn get_single_dependency_as_tuple_vector(dep: &Dependency) -> Vec<AllowedDependencyTuple> {
+    let mut tuples: Vec<AllowedDependencyTuple> = vec![];
+    for version in dep.versions.values() {
+        let name = dep.name.clone();
+        let ver = version
+            .get("version")
+            .unwrap()
```

Comment:
If it's possible for the "version" to be None, then we should account for that here and in our data structure instead of panicking.

Reply:
`version` can be None, but i think `load_allowlist` is already checking if `version` is set for an allowed dependency

```
(23-08-17 12:55:17) [~/pg_data]
$ > cat ~/allow.toml
foo = { features = ["bar"] }

(23-08-17 12:55:58) [~/pg_data]
$> psql -d postgres -c 'SELECT * FROM plrust.allowed_dependencies();'
ERROR:  Error loading dependency allow-list:
   0: Dependency entry is missing the `version` attribute

Location:
   plrust/src/allow_list.rs:432

Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.

Reply:
Yeah, I think we're fine here. A dependency is only coherent with a version of some sort, since a Rust dependency, as far as cargo is concerned, is not a crate name but also its version, so panicking  is okay.